src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java

```diff
@@ -395,19 +409,30 @@ private Object handleComplexAvroTypes(Object jsonString, Schema schema) {
         if (jsonString == null) {
             return null;
         }
+        JsonNode defaultValueNode = schema.getJsonProp("default");
+        boolean usingDefault = false;
```

Comment:
It seems the Enum type supports default value https://avro.apache.org/docs/current/spec.html. Shall we check if we have handled that too?

Reply:
Fixed with ba1167b